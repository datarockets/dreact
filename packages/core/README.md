# `@dreact/core`

> TODO: description

## Usage

```
const core = require('@dreact/core');

// TODO: DEMONSTRATE API
```
