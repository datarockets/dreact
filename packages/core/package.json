{
  "name": "dreact",
  "version": "0.10.0",
  "description": "> TODO: description",
  "homepage": "https://github.com/datarockets/dreact#readme",
  "license": "ISC",
  "bin": {
    "dreact": "bin/dreact-bridge.sh"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/datarockets/dreact.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "dependencies": {
    "next": "^10.2"
  }
}
