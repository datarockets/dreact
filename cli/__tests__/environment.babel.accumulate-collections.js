import fs from 'fs'
import path from 'path'
import { transformSync as babelTransform } from '@babel/core'
import prettier from 'prettier'

import plugin from '../environment/babel/accumulate-collections'

jest.mock('fs')

const transform = (src, options) => {
  const transformed = babelTransform(src, {
    plugins: [plugin],
    ...options,
  }).code

  return prettier
    .format(transformed, {
      filepath: __filename,
      singleQuote: true,
      semi: false,
    })
    .trim()
}

beforeAll(() => {
  // We act like we have empty ./src/collections directory by default

  fs.readdirSync.mockReturnValue([])
  fs.existsSync.mockImplementation(filepath => {
    return /src\/collections(\/\w+\/(rerducer|saga)\.js)?$/.test(filepath)
  })
})

describe('when src/collections does not exist', () => {
  beforeAll(() => {
    fs.readdirSync.mockImplementation(() => {
      throw new Error()
    })
    fs.existsSync.mockImplementation(filepath => {
      return /src\/collections$/.test(filepath) ? false : undefined
    })
  })

  it('does not do anything', () => {
    const input = 'makeStoreConfigurer()'
    const result = transform(input)

    expect(result).toBe(input)
  })
})

describe('when src/collections is empty', () => {
  beforeAll(() => {
    fs.readdirSync.mockReturnValue([])
  })

  it('does not do anything', () => {
    const input = 'makeStoreConfigurer()'
    const result = transform(input)

    expect(result).toBe(input)
  })
})

const buildCollectionDirectory = name => ({
  isDirectory: () => true,
  name,
})

describe('when src/collections has sagas', () => {
  beforeAll(() => {
    fs.readdirSync.mockReturnValue([
      buildCollectionDirectory('First'),
      buildCollectionDirectory('Second'),
    ])

    fs.existsSync.mockImplementation(filepath => {
      return /src\/collections(\/(First|Second)\/saga\.js)?$/.test(filepath)
    })
  })

  it('adds sagas property to the empty params', () => {
    const input = 'makeStoreConfigurer()'
    const result = transform(input)

    expect(result).toMatchSnapshot()
  })

  it('adds sagas to the existing param', () => {
    const input = 'makeStoreConfigurer({ sagas: ["existing"] })'
    const result = transform(input)

    expect(result).toMatchSnapshot()
  })
})

describe('when src/collections has reducers', () => {
  beforeAll(() => {
    fs.readdirSync.mockReturnValue([
      buildCollectionDirectory('First'),
      buildCollectionDirectory('Second'),
    ])

    fs.existsSync.mockImplementation(filepath => {
      return /src\/collections(\/(First|Second)\/reducer\.js)?$/.test(filepath)
    })
  })

  it('adds reducers property to the empty params', () => {
    const input = 'makeStoreConfigurer()'
    const result = transform(input)

    expect(result).toMatchSnapshot()
  })

  it('adds reducers to the existing param', () => {
    const input = 'makeStoreConfigurer({ reducers: { form: FormReducer } })'
    const result = transform(input)

    expect(result).toMatchSnapshot()
  })
})
