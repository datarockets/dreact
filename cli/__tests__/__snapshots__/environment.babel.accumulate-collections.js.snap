// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when src/collections has reducers adds reducers property to the empty params 1`] = `
"makeStoreConfigurer({
  reducers: {
    first: require('./First/reducer.js').default,
    second: require('./Second/reducer.js').default,
  },
})"
`;

exports[`when src/collections has reducers adds reducers to the existing param 1`] = `
"makeStoreConfigurer({
  reducers: {
    first: require('./First/reducer.js').default,
    second: require('./Second/reducer.js').default,
    form: FormReducer,
  },
})"
`;

exports[`when src/collections has sagas adds sagas property to the empty params 1`] = `
"makeStoreConfigurer({
  sagas: [
    require('./First/saga.js').default,
    require('./Second/saga.js').default,
  ],
})"
`;

exports[`when src/collections has sagas adds sagas to the existing param 1`] = `
"makeStoreConfigurer({
  sagas: [
    'existing',
    require('./First/saga.js').default,
    require('./Second/saga.js').default,
  ],
})"
`;
